post_processing_code/general_functions/general_post_processing.py

```diff
@@ -0,0 +1,605 @@
+            prob_under_1_prevalence_output,
+            # year that the avg prevalence is < 1%
+            year_under_1_avg_prevalence_output,
+            # year that the 90% of runs have < 1% mfp
+            year_90_under_1_prevalence_output,
+        )
+    )
+
+
+def summarize_measures(
+    raw_output_data: np.array,
+    year_column_loc: int,
+    measure_column_loc: int,
+    age_start_column_loc: int,
+    age_end_column_loc: int,
+    draws_loc: list[int],
+    prevalence_marker_name: str,
+    measure_summary_map: dict,
+) -> np.ndarray:
+    """
+    A helper function that takes in raw model outputs and summarizes requested measure values into
+    into mean, median, standard deviation, 2.5, 5, 10, 25, 50, 75, 90, 95, and 97.5 percentiles.
+
+    Args:
+        raw_output_data (pd.DataFrame): The raw data output in the form of a dataframe from multiple
+                                        runs of the model.
+        year_column_loc (int): Location of the year column.
+        measure_column_loc (int): Location of the measure column.
+        age_start_column_loc (int): Location of the age_start column.
+        age_end_column_loc (int): Location of the age_end column.
+        draws_loc list(int): Location of the draw columns.
+        prevalence_marker_name (str): The name of the prevalence measure that is used to
+                                        compare with the threhsold.
+        measure_summary_map (dict[str: function]): a map of measure names to a summary calculation
+                                                functions. This will be used to calculate the mean,
+                                                median, std, and percentiles.
+                                                By default it is `None`, which will summarize the
+                                                measure passed in `prevalence_marker_name` using
+                                                the `measure_summary_float` function.
+                                                If a measure is supplied without a callable function
+                                                `measure_summary_float` will be used by default.
+    Returns:
+        Returns a 2D Matrix with the post-processed metrics for the given input
+    """
+    # Summarizing all other outputs
+    measure_summaries_output = None
+    # If the map is empty or doesn't contain the for the prevalence marker, add it to the list of
+    # measure summaries to calculate using the default summary function
+    if (measure_summary_map is None) | (
+        prevalence_marker_name not in measure_summary_map.keys()
+    ):
+        measure_summary_map[prevalence_marker_name] = measure_summary_float
+    # replacing all `None` with the default function
+    measure_summary_map = {
+        key: (func if func is not None else measure_summary_float)
+        for key, func in measure_summary_map.items()
+    }
```

Comment:
💡 You could pull this out into a function, but I don't know what to call it, maybe `setup_minimal_measures` that would take it in a starting map, and return the completed map. It could also sanity check that every value is `callable`